src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -0,0 +1,261 @@
+package ui.components.pickers;
+
+import backend.resource.TurboLabel;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used to represent the state of the label picker. In addition to representing a state,
+ * it contains the logic that handles every state transition
+ */
+public class LabelPickerState {
+
+    private Set<String> initialLabels;
+    private List<String> addedLabels;
+    private List<String> removedLabels;
+    private List<String> matchedLabels;
+    private List<TurboLabel> allLabels;
+    private OptionalInt currentSuggestionIndex;
+
+    @SuppressWarnings("PMD")
```

Comment:
Another thing: can the suppression annotation be more specific? We might think warning x is being suppressed when in fact it could be suppressing x, y, and z!